handler/handler.go

```diff
@@ -0,0 +1,79 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/ONSdigital/dp-search-data-importer/event"
+	"github.com/ONSdigital/dp-search-data-importer/models"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out eventtest/result_writer.go -pkg eventtest . ResultWriter
+
+var _ event.Handler = (*BatchHandler)(nil)
+
+// BatchHandler handles batches of publishedContentModel events that contain CSV row data.
+type BatchHandler struct {
+	resultWriter ResultWriter
+}
+
+// TODO : ResultWriter dependency that outputs results
+type ResultWriter interface{}
+
+// NewBatchHandler returns a resultWriter TBD.
+func NewBatchHandler(
+	resultWriter ResultWriter) *BatchHandler {
+
+	return &BatchHandler{
+		resultWriter: resultWriter,
+	}
+}
+
+// Handle the given slice of PublishedContent events.
+func (batchHandler BatchHandler) Handle(ctx context.Context, events []*models.PublishedContentModel) error {
+
+	logData := log.Data{
+		"event": events,
+	}
+	log.Event(ctx, "events handler called", log.INFO, logData)
+
+	publisheContents := make([]*models.PublishedContentModel, 0, len(events))
```

Comment:
Typo? Also, you are creating an empty slice here (albeit with a capacity big enough to hold the same number of elements as the events slice) and using it later when looping over it. However you at no point copy over the contents into the slice. So there are no elements in the slice. 
```suggestion
	publishedContents := make([]*models.PublishedContentModel, 0, len(events))
```